flink-core/src/main/java/org/apache/flink/configuration/HistoryServerOptions.java

```diff
@@ -89,6 +89,16 @@
 			.withDescription("Enable HTTPs access to the HistoryServer web frontend. This is applicable only when the" +
 				" global SSL flag security.ssl.enabled is set to true.");
 
+	/**
+	 * Specifies whether should HistoryServer process job archive deletion. If this option is enabled
+	 * then deleted job archives are also deleted from HistoryServer. By default is disabled.
+	 */
+	public static final ConfigOption<Boolean> HISTORY_SERVER_PROCESS_JOB_DELETION =
+		key("historyserver.web.process-job-deletion")
```

Comment:
Can you come up with a more descriptive name? How about `historyserver.web.clean-expired-jobs`?

Reply:
That sounds much better. Thank you.